src/background/services/keyAutoAdd.ts

```diff
@@ -137,15 +143,20 @@ export class KeyAutoAddService {
           ),
         );
       } else if (message.action === 'PROGRESS') {
-        // can also save progress to show in popup
-        const currentStep = this.getCurrentStep(message.payload.steps);
+        const steps = message.payload.steps;
+        const timeoutIn = steps
+          .filter(({ status }) => status === 'pending' || status === 'active')
+          .reduce((acc, { maxDuration }) => acc + maxDuration, BASE_TIMEOUT);
+        timeout.reset(timeoutIn);
```

Comment:
This reset timeout according to steps that are remaining. But it prevents letting it go to `0` (via a large `BASE_TIMEOUT`) as we don't want to abort right after all steps are completed.